main.tf

```diff
@@ -46,3 +46,41 @@ module "s3_user" {
   s3_actions   = ["${var.allowed_bucket_actions}"]
   s3_resources = ["${join("", aws_s3_bucket.default.*.arn)}/*", "${join("", aws_s3_bucket.default.*.arn)}"]
 }
+
+resource "aws_s3_bucket_policy" "default" {
+  count  = "${var.allow_encrypted_uploads_only == "true" ? 1 : 0}"
+  bucket = "${aws_s3_bucket.default.id}"
+
+  policy = <<POLICY
```

Comment:
use https://www.terraform.io/docs/providers/aws/d/iam_policy_document.html
Example: https://github.com/cloudposse/terraform-aws-eks-cluster/blob/master/main.tf#L12

Reply:
why not to use `aws_s3_bucket_policy`?

Reply:
it's our best practice to use a policy document TF resource and not HEREDOC.
HEREDOC is just a string with no type, and more difficult to read and maintain

Reply:
done